File: debug-xiaohongshu.py
#!/usr/bin/env python3
"""
调试小红书URL解析过程，特别是实况图片提取逻辑
"""
import re
import json
import httpx
from media_parser_sdk.platforms.xiaohongshu import XiaohongshuParser

def debug_xiaohongshu_parsing(url):
    """调试小红书URL解析过程"""
    print(f"=== 调试小红书URL解析: {url} ===")
    
    parser = XiaohongshuParser()
    
    try:
        # 获取网页HTML
        print("\n1. 获取网页HTML...")
        headers = {
            "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
            "Referer": "https://www.xiaohongshu.com/",
            "Accept-Language": "zh-CN,zh;q=0.9"
        }
        
        with httpx.Client(headers=headers, timeout=15, follow_redirects=True) as client:
            response = client.get(url)
            response.raise_for_status()
            html = response.text
        
        print(f"HTML长度: {len(html)} 字符")
        
        # 提取__INITIAL_STATE__
        print("\n2. 提取__INITIAL_STATE__...")
        initial_state_pattern = re.compile(r'window\.__INITIAL_STATE__\s*=\s*(.+?)(?=</script>)', re.DOTALL)
        initial_state_match = initial_state_pattern.search(html)
        
        if initial_state_match:
            initial_state_str = initial_state_match.group(1).strip()
            if initial_state_str.endswith(';'):
                initial_state_str = initial_state_str[:-1]
            
            print(f"__INITIAL_STATE__长度: {len(initial_state_str)} 字符")
            
            try:
                # 尝试解析JSON
                initial_state = json.loads(initial_state_str)
                print("✅ 成功解析__INITIAL_STATE__")
                
                # 检查noteDetailMap
                note = initial_state.get("note", {})
                note_detail_map = note.get("noteDetailMap", {})
                print(f"noteDetailMap 键数量: {len(note_detail_map.keys())}")
                
                if note_detail_map:
                    note_id = next(iter(note_detail_map.keys()), None)
                    if note_id:
                        note_detail = note_detail_map[note_id]
                        note_data = note_detail.get("note", {})
                        print(f"note_data 存在: {bool(note_data)}")
                        
                        # 检查imageList
                        image_list = note_data.get("imageList", [])
                        print(f"imageList 长度: {len(image_list)}")
                        
                        # 检查每张图片是否包含livePhoto
                        has_live_photo = False
                        for i, img in enumerate(image_list):
                            live_photo = img.get("livePhoto")
                            print(f"  图片 {i+1}: livePhoto = {live_photo}")
                            if live_photo:
                                has_live_photo = True
                                print(f"    ✅ 发现实况图片: {live_photo}")
                                
                                # 检查livePhoto的结构
                                if isinstance(live_photo, dict):
                                    print(f"    livePhoto类型: dict")
                                    print(f"    livePhoto键: {list(live_photo.keys())}")
                                    print(f"    videoUrl存在: {bool(live_photo.get('videoUrl'))}")
                        
                        print(f"\n3. 提取实况图片结果:")
                        print(f"   检测到实况图片: {has_live_photo}")
            except json.JSONDecodeError as e:
                print(f"❌ JSON解析失败: {e}")
                print(f"原始数据前1000字符: {initial_state_str[:1000]}...")
        else:
            print("❌ 未找到__INITIAL_STATE__")
            
        # 检查页面中是否包含livePhoto相关内容
        print("\n4. 检查页面中livePhoto相关内容...")
        live_photo_matches = re.findall(r'livePhoto', html)
        print(f"页面中livePhoto出现次数: {len(live_photo_matches)}")
        
        if live_photo_matches:
            # 提取包含livePhoto的上下文
            context_matches = re.findall(r'.{0,100}livePhoto.{0,100}', html, re.DOTALL)
            print(f"提取到 {len(context_matches)} 个上下文片段")
            for i, match in enumerate(context_matches[:3]):
                print(f"  片段 {i+1}: {match[:200]}...")
    
    except Exception as e:
        print(f"❌ 调试过程中发生错误: {str(e)}")
        import traceback
        traceback.print_exc()

def main():
    """主函数"""
    # 提示用户输入小红书URL
    test_url = input("请输入小红书URL: ").strip()
    if not test_url:
        print("URL不能为空")
        return
    
    debug_xiaohongshu_parsing(test_url)

if __name__ == "__main__":
    main()

File: backend/debug-xiaohongshu.js
const axios = require('axios');
const cheerio = require('cheerio');
const fs = require('fs-extra');
const path = require('path');

async function debugXiaohongshu(url) {
  console.log('=== 小红书链接调试分析 ===');
  console.log(`🔗 目标URL: ${url}`);
  
  try {
    console.log('📡 开始获取页面内容...');
    
    const headers = {
      'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36',
      'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
      'Accept-Language': 'zh-CN,zh;q=0.9,en;q=0.8',
      'Connection': 'keep-alive'
    };
    
    const response = await axios.get(url, {
      headers,
      timeout: 15000
    });
    
    console.log(`✅ 页面获取成功，状态码: ${response.status}`);
    console.log(`📄 内容长度: ${response.data.length} 字符`);
    
    const html = response.data;
    const $ = cheerio.load(html);
    
    // 基本信息
    const title = $('title').text();
    console.log(`📝 页面标题: ${title}`);
    
    // 查找JSON数据
    console.log('🔍 查找JSON数据...');
    
    const initialStateMatch = html.match(/window\.__INITIAL_STATE__\s*=\s*(\{[\s\S]*?\})(?=\s*;|\s*<\/script>)/);
    
    if (initialStateMatch) {
      console.log('✅ 找到 __INITIAL_STATE__ 数据');
      
      try {
        let jsonStr = initialStateMatch[1];
        jsonStr = jsonStr.replace(/:\s*undefined\s*(,|\}|\])/g, ': null$1');
        jsonStr = jsonStr.replace(/,\s*(\}|\])/g, '$1');
        
        const jsonData = JSON.parse(jsonStr);
        console.log(`✅ JSON解析成功`);
        console.log(`🔑 顶级键: ${Object.keys(jsonData).slice(0, 10).join(', ')}`);
        
        // 查找note数据
        let noteData = null;
        
        if (jsonData.note && jsonData.note.noteDetailMap) {
          const noteIds = Object.keys(jsonData.note.noteDetailMap);
          if (noteIds.length > 0) {
            noteData = jsonData.note.noteDetailMap[noteIds[0]].note;
            console.log('✅ 找到note数据');
          }
        }
        
        if (noteData) {
          console.log(`📝 标题: ${noteData.title || '未找到'}`);
          console.log(`👤 作者: ${noteData.user?.nickname || '未找到'}`);
          console.log(`📄 描述: ${(noteData.desc || '未找到').substring(0, 100)}`);
          
          if (noteData.imageList && Array.isArray(noteData.imageList)) {
            console.log(`🖼️  图片数量: ${noteData.imageList.length}`);
            
            noteData.imageList.forEach((img, index) => {
              console.log(`图片 ${index + 1}:`);
              console.log(`  url_default: ${img.url_default || '无'}`);
              console.log(`  url_pre: ${img.url_pre || '无'}`);
              console.log(`  url: ${img.url || '无'}`);
              
              if (img.stream) {
                console.log(`  包含视频流数据`);
              }
            });
          } else {
            console.log('❌ 未找到imageList');
          }
        } else {
          console.log('❌ 未找到note数据');
        }
        
      } catch (parseError) {
        console.log(`❌ JSON解析失败: ${parseError.message}`);
      }
    } else {
      console.log('❌ 未找到 __INITIAL_STATE__ 数据');
    }
    
  } catch (error) {
    console.error('❌ 调试过程中发生错误:', error.message);
    
    if (error.response) {
      console.log(`📡 HTTP状态: ${error.response.status}`);
    }
  }
}

// 运行调试
if (require.main === module) {
  const testUrl = process.argv[2] || 'https://www.xiaohongshu.com/explore/694269d1000000001f00dc48';
  
  debugXiaohongshu(testUrl)
    .then(() => {
      console.log('🎉 调试完成！');
    })
    .catch((error) => {
      console.error('调试失败:', error);
    });
}

module.exports = { debugXiaohongshu };